nemo/collections/llm/api.py

```diff
@@ -249,6 +250,99 @@ def validate(
+        algorithm=algorithm,
+        awq_block_size=awq_block_size,
+        sq_alpha=sq_alpha,
+        enable_kv_cache=enable_kv_cache,
+        calibration_dataset=calibration_dataset,
+        calibration_dataset_size=calibration_dataset_size,
+        calibration_batch_size=calibration_batch_size,
+        calibration_seq_len=calibration_seq_len,
+    )
+
+
+@run.cli.factory
+@run.autoconvert
+def default_export(
+    path: str,
+    dtype: Union[str, int] = "bf16",
+    decoder_type: Optional[str] = None,
+    inference_tensor_parallel: int = 1,
+    inference_pipeline_parallel: int = 1,
+) -> ExportConfig:
+    """Default export configuration."""
+    return ExportConfig(
+        path=path,
+        dtype=dtype,
+        decoder_type=decoder_type,
+        inference_tensor_parallel=inference_tensor_parallel,
+        inference_pipeline_parallel=inference_pipeline_parallel,
+    )
+
+
+@run.cli.entrypoint(name="ptq", namespace="llm")
+def ptq(
+    nemo_checkpoint: str,
+    calib_tp: int = 1,
+    calib_pp: int = 1,
+    quantization_config: QuantizationConfig = default_quantization(),
+    export_config: ExportConfig = default_export(None),
+):
+    """
+    Applies Post-Training Quantization (PTQ) for a model using the specified quantization and export configs. It runs
+    calibration for a small dataset to collect scaling factors low-precision GEMMs used by desired quantization method.
+
+    This function produces TensorRT-LLM checkpoint ready for deployment using nemo.export and nemo.deploy modules
+    or direcly using TensorRT-LLM library.
+
+    Args:
+        nemo_checkpoint (str): The path to model to be quantized.
+        calib_tp (int): Calibration tensor parallelism.
+        calib_pp (int): Calibration pipeline parallelism.
+        quantization_config (QuantizationConfig): Configuration for quantization algorithm.
+        export_config (ExportConfig): Export configuration for TensorRT-LLM checkpoint.
+
+    Returns:
+        Path: The directory path where quantized model is saved.
+
+    Example:
+        >>> from nemo.collections.llm.quantization import ExportConfig, QuantizationConfig
+        >>> nemo_checkpoint = "/opt/checkpoints/LLAMA3-8B-fp16"
+        >>> quantization_config = QuantizationConfig(algorithm="fp8")
+        >>> export_config = ExportConfig(path="/opt/checkpoints/LLAMA3-8B-fp8")
+        >>> llm.ptq(nemo_checkpoint, quantization_config=quantization_config, export_config=export_config)
+        '/opt/checkpoints/LLAMA3-8B-fp8'
+    """
+    if export_config.path is None:
+        raise ValueError("The export_config.path needs to be specified, got None.")
+
+    from nemo.collections.llm import quantization
+
+    quantizer = quantization.Quantizer(quantization_config, export_config)
+
+    model = quantization.load_with_modelopt_layer_spec(nemo_checkpoint, calib_tp, calib_pp)
+
+    model = quantizer.quantize(model)
+
+    quantizer.export(model)
+
+    return export_config.path
```

Comment:
Let's print the path + success message, like we do [here](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/llm/api.py#L476-L481)

Reply:
Addressed in https://github.com/NVIDIA/NeMo/pull/10984/commits/88bee68ad686155653ceefa119e9a907509d07e6